Adds seed derivatives command & UI

typhonius opened the pull request.

typhonius committed:
Adds first commit of seed derivatives command

saltednut commented on modules/df/df_tools/df_tools_image/src/Form/CreateDerivativesForm.php:
```diff
@@ -0,0 +1,124 @@
+    );
+    $form['run'] = array(
+      '#type' => 'submit',
+      '#value' => t('Create Derivatives'),
```
This will be great for SA's.

typhonius committed:
Makes the user interface work for image derivative generation.

typhonius commented:
@brantwynn I _think_ I've fixed the user interface so this can now be used to create derivatives if the user does not have access to Drush.

saltednut reviewed (commented):
This is looking good at least from a quick read through. I will say, it looks like there is enough going on in the .module file that you're probably better off moving everything into a service. The nice thing about services is that they can still be loaded by the global container for Drush commands, but you should be able to clean up instances of \Drupal throughout the global functions as you convert them over to methods inside a handler. For reference, check out Scenarios module's handler under modules/contrib/scenarios/src/scenariosHandler.phpThe main reason I bring this up is that the current code still looks somewhat stopgap from 7 to 8 rather than native D8 code and I myself have written a lot of this style stuff, but when I finally decided to do it "the drupal 8 way" I learned a ton along the way.

saltednut commented on modules/df/df_tools/df_tools_image/df_tools_image.permissions.yml:
```diff
@@ -0,0 +1,4 @@
+administer auto login url:
```
Looks like a quick fix :)

saltednut commented:
For the drush command, it does work. The count does not initially, so it says "  entries are to have their image derivatives created." with the variable string missing. Also tested via UI and that works.

As I said before, I'd like to see the .module turned into a proper service, but thats not really a blocker for merging.  We should definitely fix the permissions declaration as it currently says "administer auto login url" - thanks!

typhonius committed:
Allows users to specify which derivatives they want to create via the UI.

typhonius committed:
Fixes permission name.

typhonius committed:
Changes the placeholder so drush output prints number of entries.

typhonius commented:
Thanks @brantwynn for the advice about making this a service. Definitely something I'd be keen to learn more about. I've fixed the drush output, the permissions name and added in checkboxes. If you'd like me to collapse this all into a single commit, I'm happy to do that.

typhonius committed:
Add in a description around how checkboxes should be used.

saltednut commented:
@typhonius np. We can merge this in as-is and if you have time we can chat about services and do a follow up PR later. I can squash it on your behalf and merge, but its up to you. This looks good on my end though. @briancwald did you want to do a quick run through?
